File: ShapesPainter/app/src/main/java/com/example/alexander/shapespainter/Rectangle.java
package com.example.alexander.shapespainter;

import android.graphics.Color;

/**
 * Created by Alexander on 19.01.2017.
 */

public class Rectangle extends Shape {
    private Color mFillColor;
    private Color mOutlineColor;
    private Vector2f mLeftTop;
    private VectorSize2f mSize;

    Rectangle(Vector2f leftTop, VectorSize2f size, Color fillColor, Color outlineColor) {
        mLeftTop = leftTop;
        mSize = size;
        mFillColor = fillColor;
        mOutlineColor = outlineColor;
    }

    @Override
    void draw(ICanvas canvas) {
        canvas.setFillColor(mFillColor);
        canvas.setOutlineColor(mOutlineColor);
        canvas.drawRectangle(mLeftTop, mSize, mFillColor, mOutlineColor);
    }

    @Override
    public ShapeType getType() {
        return null;
    }

    @Override
    public ShapeDiagram getDiagram() {
        return null;
    }

    @Override
    public Color getFillColor() {
        return null;
    }

    @Override
    public Color getOutlineColor() {
        return null;
    }

    @Override
    public Vector2f getVertices() {
        return null;
    }


    @Override
    public boolean isPointInside(Vector2f point) {
        return false;
    }

    @Override
    public Vector2f getCenter() {
        return null;
    }

    @Override
    public Vector2f getSize() {
        return null;
    }
}

File: Fabric/app/src/main/java/com/example/alexander/fabric/DesignerTest/Rectangle.kt
package com.example.alexander.fabric.DesignerTest
import com.example.alexander.fabric.Definitions.ColorEnum
import com.example.alexander.fabric.Definitions.CordPoint
import com.example.alexander.fabric.Interfaces.ICanvas
import com.example.alexander.fabric.Interfaces.Shape

/**
 * Created by Alexander on 25.06.2016.
 */
class Rectangle(var leftTop: CordPoint, val rightBottom: CordPoint, color : ColorEnum) : Shape(color)
{
    override fun Draw(canvas: ICanvas)
    {
        val x1 = leftTop.x
        val y1 = leftTop.y
        val x2 = rightBottom.x
        val y2 = rightBottom.y
        canvas.SetColor(color)
        canvas.DrawLine(CordPoint(x1, y1), CordPoint(x1, y2))
        canvas.DrawLine(CordPoint(x1, y2), CordPoint(x2, y2))
        canvas.DrawLine(CordPoint(x2, y2), CordPoint(x2, y1))
        canvas.DrawLine(CordPoint(x2, y1), CordPoint(x1, y1))
    }
    override fun toString(): String {
        return "Rectangle: color: ${color.name} data: ${leftTop.toString()}, ${rightBottom.toString()}"
    }
}